components/src/core/Drawer/DrawerNavGroup.tsx

```diff
@@ -112,31 +188,146 @@ function NavigationListItem(item: NavItem, props: DrawerNavGroupProps): ReactNod
         }
     };
 
+    const chevron = groupChevron === undefined ? itemChevron : groupChevron;
+
+    function getRightActionIcon(): JSX.Element {
+        if (item.rightComponent) {
+            return item.rightComponent;
+        }
+        if (chevron) {
+            return null;
+        }
+        if (!subItems) {
+            return null;
+        }
+        if (depth) {
+            return <ArrowDropUp />;
+        }
+        return <ExpandLess />;
+    }
+
+    const rightActionIcon = getRightActionIcon();
+
+    const rightAction = (): void => {
+        if (!rightActionIcon) return;
+        if (onClickIcon) {
+            onClickIcon();
+        } else {
+            action();
+        }
+    };
```

Comment:
See comment above about joining the expand icon with anything passed to rightComponent